6.5.group_by.py

```diff
@@ -0,0 +1,14 @@
+def group_by(func, *iterable):
```

Comment:
If you return a complicated statement you have to use type hints and return types -
https://www.geeksforgeeks.org/duck-typing-in-python/
https://stackoverflow.com/questions/125367/dynamic-type-languages-versus-static-type-languages
https://docs.python.org/3/library/typing.html
https://realpython.com/lessons/pros-and-cons-type-hints/